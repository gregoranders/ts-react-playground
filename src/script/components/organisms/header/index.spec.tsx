import React, { mount, create } from '@app/testUtils';

import { MemoryRouter } from 'react-router-dom';

import * as TestSubject from '@organisms/header';

describe(`${TestSubject.Header.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <MemoryRouter>
        <TestComponent />
      </MemoryRouter>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <MemoryRouter>
        <TestComponent />
      </MemoryRouter>,
    );
    expect(testSubject.text()).toBe('Header');
  });
});
