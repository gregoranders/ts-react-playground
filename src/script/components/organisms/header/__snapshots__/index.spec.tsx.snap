// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<header
  className="w3-center"
>
  <h1
    className="w3-jumbo w3-wide"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Header
    </a>
  </h1>
</header>
`;
